core/src/test/java/edu/ucr/cs/riple/core/AnalysisModeTest.java

```diff
@@ -0,0 +1,88 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.core;
+
+import static edu.ucr.cs.riple.core.AnalysisMode.LOWER_BOUND;
+import static edu.ucr.cs.riple.core.AnalysisMode.STRICT;
+import static edu.ucr.cs.riple.core.Report.Tag.APPROVE;
+import static edu.ucr.cs.riple.core.Report.Tag.REJECT;
+
+import edu.ucr.cs.riple.core.tools.TReport;
+import edu.ucr.cs.riple.injector.location.OnMethod;
+import java.util.List;
+import java.util.Objects;
+import org.junit.Test;
+
+public class AnalysisModeTest extends BaseCoreTest {
+
+  public AnalysisModeTest() {
+    super("analysis-mode-test", List.of("Target", "Dep"));
+  }
+
+  @Test
+  public void strictModeTest() {
+    coreTestHelper
+        .addExpectedReports(
+            // Resolves 6 errors locally and all errors on downstream dependencies can be resolved
+            // with no new triggered error, therefore the effect is -6. The fix triggers no error
+            // in downstream dependencies, should be approved.
+            new TReport(
+                new OnMethod("Foo.java", "test.target.Foo", "returnNullGood()"), -6, APPROVE),
+            // Resolves 6 errors locally but creates 1 error on downstream dependencies that cannot
+            // be resolved, therefore the effect is -5. The fix triggers 1 error in downstream
+            // dependencies, should be rejected.
+            new TReport(new OnMethod("Foo.java", "test.target.Foo", "returnNullBad()"), -5, REJECT))
+        .setPredicate(
+            (expected, found) ->
+                expected.root.equals(found.root)
+                    && expected.getExpectedValue()
+                        == found.getOverallEffect(coreTestHelper.getConfig())
+                    && Objects.equals(expected.getTag(), found.getTag()))
+        .toDepth(5)
+        .disableBailOut()
+        .enableDownstreamDependencyAnalysis(STRICT)
+        .start();
+  }
+
+  @Test
+  public void lowerBoundModeTest() {
+    coreTestHelper
+        .addExpectedReports(
+            new TReport(
+                new OnMethod("Foo.java", "test.target.Foo", "returnNullGood()"), -6, APPROVE),
+            new TReport(
+                new OnMethod("Foo.java", "test.target.Foo", "returnNullBad()"), -5, APPROVE))
+        .setPredicate(
+            (expected, found) ->
+                expected.root.equals(found.root)
+                    && expected.getExpectedValue()
+                        == found.getOverallEffect(coreTestHelper.getConfig())
+                    && Objects.equals(expected.getTag(), found.getTag()))
+        .toDepth(5)
+        .disableBailOut()
+        .enableDownstreamDependencyAnalysis(LOWER_BOUND)
+        .start();
+  }
+}
```

Comment:
Do we want an upper bound test? (That will need a second dependency, but it's probably a good idea to check that lower/upper bound do what they are supposed to do when there is more than one dependency)